src/components/MenuModal/MenuModal.component.jsx

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect, useCallback, useRef } from 'react';
+import { useHistory } from 'react-router-dom';
+import Styled from './styled';
+import { useAppContext } from '../../context/AppProvider';
+
+const MenuModal = () => {
+  const { state } = useAppContext();
+  const { isLogged } = state;
+  const outRef = useRef(null);
+  const history = useHistory();
+
+  const goBack = useCallback(() => {
+    history.goBack();
+  }, [history]);
+
+  // Added functionality to close modal when pressing ESC key.
+  const escFunction = useCallback(
+    (event) => {
+      if (event.keyCode === 27) {
+        goBack();
+      }
+    },
+    [goBack]
+  );
+
+  useEffect(() => {
+    document.addEventListener('keydown', escFunction, false);
+
+    return () => {
+      document.removeEventListener('keydown', escFunction, false);
+    };
+  }, [escFunction]);

```

Comment:
Definitely this should go into a util or a custom hook. As you can see, this is the third time this code is being copy & pasted, which is a bad practice. Whenever you have to copy & paste code to have the same functionality that's a good indicator that you should consider using a util.